### ISSUE TITLE
History font size on Linux

### ISSUE THREAD

**Issue by akontsevich** [gitahead/gitahead#297]
Cool Git GUI - thanks!!! Wish it rapid development! 

For now it is possible to setup font only for Editor (Tools->Options->Editor). Font in History remain very small and uncomfortable. Please add option to setup font there as well.

**Comment by hackhaslam** [comment]
Can you post a screenshot of what you're seeing? It sounds like a bug.

**Comment by akontsevich** [comment]
Ha, can't do, started to crash:
```
 GitAhead
Lua Error: module 'lexer' not found:
        no field package.preload['lexer']
        no file '/home/abuild/rpmbuild/BUILD/gitahead-2.5.10/dep/scintilla/scintilla-3.10.6/lexlua/lexer.lua'
        no file '/usr/local/lib/lua/5.3/lexer.so'
        no file '/usr/local/lib/lua/5.3/loadall.so'
        no file './lexer.so'.
Lua Error: module 'lexer' not found:
        no field package.preload['lexer']
        no file '/home/abuild/rpmbuild/BUILD/gitahead-2.5.10/dep/scintilla/scintilla-3.10.6/lexlua/lexer.lua'
        no file '/usr/local/lib/lua/5.3/lexer.so'
        no file '/usr/local/lib/lua/5.3/loadall.so'
        no file './lexer.so'.
Lua Error: module 'lexer' not found:
        no field package.preload['lexer']
        no file '/home/abuild/rpmbuild/BUILD/gitahead-2.5.10/dep/scintilla/scintilla-3.10.6/lexlua/lexer.lua'
        no file '/usr/local/lib/lua/5.3/lexer.so'
        no file '/usr/local/lib/lua/5.3/loadall.so'
        no file './lexer.so'.
Lua Error: module 'lexer' not found:
        no field package.preload['lexer']
        no file '/home/abuild/rpmbuild/BUILD/gitahead-2.5.10/dep/scintilla/scintilla-3.10.6/lexlua/lexer.lua'
        no file '/usr/local/lib/lua/5.3/lexer.so'
        no file '/usr/local/lib/lua/5.3/loadall.so'
        no file './lexer.so'.
Lua Error: module 'lexer' not found:
        no field package.preload['lexer']
        no file '/home/abuild/rpmbuild/BUILD/gitahead-2.5.10/dep/scintilla/scintilla-3.10.6/lexlua/lexer.lua'
        no file '/usr/local/lib/lua/5.3/lexer.so'
        no file '/usr/local/lib/lua/5.3/loadall.so'
        no file './lexer.so'.
Lua Error: module 'lexer' not found:
        no field package.preload['lexer']
        no file '/home/abuild/rpmbuild/BUILD/gitahead-2.5.10/dep/scintilla/scintilla-3.10.6/lexlua/lexer.lua'
        no file '/usr/local/lib/lua/5.3/lexer.so'
        no file '/usr/local/lib/lua/5.3/loadall.so'
        no file './lexer.so'.
Lua Error: module 'lexer' not found:
        no field package.preload['lexer']
        no file '/home/abuild/rpmbuild/BUILD/gitahead-2.5.10/dep/scintilla/scintilla-3.10.6/lexlua/lexer.lua'
        no file '/usr/local/lib/lua/5.3/lexer.so'
        no file '/usr/local/lib/lua/5.3/loadall.so'
        no file './lexer.so'.
PANIC: unprotected error in call to Lua API (attempt to index a nil value)
Аварийный останов (стек памяти сброшен на диск)
```

**Comment by akontsevich** [comment]
RPM crashes, your `.sh` works fine. Screenshot:

![Screenshot_20191018_234438](https://user-images.githubusercontent.com/3843657/67127102-4fd8ad00-f201-11e9-95f5-f6038f83d80c.png)

**Comment by akontsevich** [comment]
Another bug in UI - buttons overlap text:

![Screenshot_20191019_001233](https://user-images.githubusercontent.com/3843657/67128732-3f2a3600-f205-11e9-82f8-b50ed46ff350.png)

**Comment by hackhaslam** [comment]
Neither of those things are normal or expected. What distro, window manager, and font scaling settings are you using?

**Comment by akontsevich** [comment]
openSUSE Tumbleweed, KDE latest, font scaling 1.0 I guess, DPI 52" Full HD

**Comment by hackhaslam** [comment]
I'm not able to reproduce that with a new install of openSUSE with KDE. I see this:

![screencap](https://user-images.githubusercontent.com/6586470/67426659-cb3ab580-f597-11e9-9e0c-8c94bbd76691.png)

**Comment by akontsevich** [comment]
I have 52 DPI, 42" Full HD screen. Many apps fails with this. Could You just add font settings for the history please?

**Comment by cata0309** [comment]
The UI font is tiny, how do I change that  ?

**Comment by akontsevich** [comment]
Could You make it the same size as in editor, for ex?

**Comment by hackhaslam** [comment]
@cata0309 Is it only the font in the history view that is small, like in the screenshot from @akontsevich? Is your environment similar to his?

**Comment by hackhaslam** [comment]
@akontsevich I understand that this is a real problem, but I don't think that making it the same size as the editor font is appropriate. The editor font is a special case. It's expected to be a different fixed-width font.

I will consider adding a separate setting for the history view that's only accessible from the command line or something similar. I would really rather understand and fix the bug instead of papering over it.

**Comment by akontsevich** [comment]
>I will consider adding a separate setting for the history view that's only accessible from the command line or something similar. 

Not convenient!

>I would really rather understand and fix the bug instead of papering over it.

Bug is very simple: on some screens DPI fonts sizes are still incorrect - Qt and KDE issue. Hope they've fixed that in recent versions (Qt 5.14-15, etc). So to make it work everywhere need to set this font size not fixed but proportional to menu font size which set by KDE settings and has suitable to user size. Or allow user to set it.

**Comment by hackhaslam** [comment]
> Not convenient!

Right, it's not meant to be convenient. It's a workaround that shouldn't be necessary.

> Bug is very simple: on some screens DPI fonts sizes are still incorrect - Qt and KDE issue. Hope they've fixed that in recent versions (Qt 5.14-15, etc). So to make it work everywhere need to set this font size not fixed but proportional to menu font size which set by KDE settings and has suitable to user size.

If that's the issue then I can fix it by making font sizes relative. Simple.

**Comment by akontsevich** [comment]
> If that's the issue then I can fix it by making font sizes relative to the application font. Simple.

Thanks! That would be good.  Many applications has this problems: some very small fonts, some (Electron/GTK) - too big. I have option whether to set 96 DPI to my screen manually and have all the fonts mostly correct but font sizes on the screen - not, or set real - 52 DPI and still have some issues in some app where fonts sizing change is not supported.

**Comment by cata0309** [comment]
> @cata0309 Is it only the font in the history view that is small, like in the screenshot from @akontsevich? Is your environment similar to his?

Yes, it's similar but the whole UI is small, including the menus text and the history view, the editor is good ! 
It would be a great feature to add an option to control the size and also the font for the UI.

**Comment by hackhaslam** [comment]
This should be fixed in 2.6.0.

**Comment by akontsevich** [comment]
> This should be fixed in 2.6.0.

Thanks much, will check!

**Comment by prasannakog** [comment]
I'm using 2.6.1 on linux and can't change the UI font size.

**Comment by akkiss** [comment]
I'm using 2.6.1 on Ubuntu 16 and all texts are so small that I can hardly read - including menu, history, diff file list etc, except for diff for which I can set the font.

The problem seems to be with Screen settings / Qt because if I turn off automatic screen scaling in Qt it shows ok:

> QT_AUTO_SCREEN_SCALE_FACTOR=0 ./GitAhead

**Comment by sbigaret** [comment]
Hi, I had a similar problem, the fonts' size not following the default LF (debian 10, wm is e16, + mate)
I solved this with `QT_SCALE_FACTOR=1.2 ./GitAhead` (give it a try and adapt the scale factor to your taste)
HTH !
